include/grpc/event_engine/endpoint_config.h

```diff
@@ -30,11 +30,21 @@ namespace experimental {
 /// document its set of supported configuration options.
 class EndpointConfig {
  public:
-  virtual ~EndpointConfig() = default;
-  using Setting = absl::variant<absl::monostate, int, absl::string_view, void*>;
+  class OptionsAccessor;
+  EndpointConfig();
+  explicit EndpointConfig(std::unique_ptr<OptionsAccessor> impl);
+  ~EndpointConfig();
+  /// Prevent copy.
+  EndpointConfig(const EndpointConfig&) = delete;
+  EndpointConfig& operator=(const EndpointConfig&) = delete;
+  using Setting =
+      absl::variant<absl::monostate, int, absl::string_view, bool, void*>;
```

Comment:
Agreed. A bool variant was intentionally left out due to its implicit construction wonkiness, and some backwards incompatible changes made because of it. https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0608r3.html.

Reply:
Removed the boolean.